test/ext/flux/flux.jl

```diff
@@ -0,0 +1,40 @@
+using Pkg
+Pkg.activate(@__DIR__)
+Pkg.develop(; path=joinpath(@__DIR__, "..", "..", ".."))
+
+using Mooncake, StableRNGs, Test
+using Flux: Losses
+using Mooncake.TestUtils: test_rule
+
+@testset "flux" begin
+    @testset "$f, $(typeof(fargs))" for (
+        interface_only, perf_flag, is_primitive, f, fargs...
+    ) in vcat(
+        # Testing specific floating point precisions as FDM is unstable for certain tests.
+        # Larger differences can compare well with AD for for higher precisions as FDM is relatively stable.
+        # for smaller differences we start getting better FDM gradients for smaller floating points.
+
+        map([Float64]) do P
+            return (
+                false, :none, false, Losses.mse, P.([1e1, 1e2, 1e3]), P.([1e3, 1e4, 0])
```

Comment:
```suggestion
                false, :none, true, Losses.mse, randn(StableRNG(1), P, 3), randn(StableRNG(2), P, 3))
```
Since this is now a primitive, we should tweak the corresponding flag to ensure that the `test_rule` checks to see whether it's hitting the primitive. The same applies to the other test cases in this file. Also, I'd really recommend just generating random numbers here, but using a seeded `StableRNG`, in order to be consistent with the other tests in this package.

The same line of reasoning applies to the other test cases in this file.

Incidentally, it looks like this extension isn't loading properly (see the Flux extension). I suspect that CI will pick up on this once you change the `is_primitive` flag to `true`.

Reply:
cool then i think all the test cases would be in just one map(), but it wouldnt show the Float precisions failing at certain cases. Would that be okay?

Reply:
Yeah, that's completely fine -- a single `map` makes sense to me.

Reply:
To be honest, I would just test on `Float32` and `Float64`.

Reply:
perfect ill do it. Thanks.